### PR TITLE
[TM NEEDED] [REVIEW NEEDED] Medical Rework MKII

### PR DESCRIPTION
**Pull request by AffectedArc07**
## About The Pull Request
Reworks medical to improve organ damage and allow bone breaking

# Bone Breaking
Bones now break when hit with excessively blunt objects (High force). They need to be healed with surgery (Drapes, Scalpel, Retractor, Hemostat, Cautery). A handheld medical scanner will tell you if a subject has broken bones, but not which ones. A full size medical scanner will tell you the location of the broken bones, as well as organ healths.

# Organ Damage
Organs now decay when not in a living host. Defibs thus have a loose timer for when a defibrillation can occur, but are dependent on things such as your heart/brain damage, which will decay at differing rates. As it is now, the heart decays fully in about 2.5 minutes, and the brain decays a lot slower, at around 30 minutes. The brain damage done to simulate decay for defibrillation was removed as a result. Furthermore, brain failure is caused when reaching max brain damage, or forceably removing the brain from its head. This also means that you can fix brains which reach their max damage thresholds by either pouring mannitol onto it, or by performing brain surgery on it.

The PR has currently:
- Extended defib timer from 2 minutes to 15 minutes
- Organ decay, with the standard decay reaching failing levels at about 15 minutes of decay
- The heart decays far faster, reaching failing levels at 3 minutes
- The brain decays twice as slow for balance reasons, since it still carries traumas over on cloning
- The heart and lungs both increase the rate at which you breath when severely damaged
- A damaged stomach will cause you to vomit depending on how much food you've eaten, and how badly damaged your stomach is. A failing one prevents digestion.
- A failing appendix will burst, causing toxic damage until removed. Appendicitis causes appendix damage when at its max stage.
- Damaged ears will cause bouts of tinnitus, which deafens the individual for a brief four second period as the ringing in his ears grows louder.
- Messages for when the above are damaged/failing to the owner
- Brain damage was moved to work off of brain.damage instead of brain.brainloss
- Removed decay factor from synthetic organs
- Strange Reagent heals all organs before revive, though that is subject to change (some organs may remain damaged)
- Freezers, coolers, and morgue units all prevent organ decay from objects held within
- Lobectomy, a surgical procedure that involves excising a decaying/damaged lobe within the lungs so that it no longer hinders its performance. Can only be done if lung damage is above 80. Sets the damage to 80, and clears the "failing" flag, making the organ usable again. Can only be done once per organ. 95% chance to succeed on the lobectomy step. Failing to perform the lobectomy will suffocate the patient momentarily as blood collects in their lungs.
- Coronary Bypass, a surgical procedure that does the same as the above surgery. 90% to succeed on both incising into the heart and grafting the bypass. Failing either step will cause damage to the heart, and massive amounts of bleeding.
- Organ Smartfridges, an organ storage device (stores significantly less than other smartfridges, 20 parts per tier of matter bin), BUT the device also repairs organs when upgraded beyond the first tier. Every tier beyond the first will repair organs by a magnitude of the normal healing rate, up to 7 times as fast as normal healing. This healing will not fix broken organs, however.
Organs heal at a rate improved by your healthiness

# Medical Scanners
In medbay, as well as the organ smartfridges, are medical scanners. These are used to show how damaged organs are, and if bones are broken. Image below:
image

## TLDR
Organs can be damaged, and will decay if not in a fridge or a living host
Defib timer is a lot longer but depends on condition of organs
Bones can break now, surgery to fix is Drapes, Scalpel, Retractor, Hemostat, Cautery
Lobectomy: Fixes lungs below 80 damage. Surgery is: Drapes, Scalpel, Retractor, Bone Saw, Hemostat, Scalpel, Cautery
Coronary Bypass: Same as the above but for the heart. Surgery is Drapes, Scalpel, Retractor, Bone saw, Hemostat, Scalpel, Hemostat, Cautery

## Why It's Good For The Game
I want a medical system that isnt total shit

## Changelog
:cl: AffectedArc07, Ported from TG and Oracle
add: Bones now break
add: Medical scanners are now a thing. Use them to look at broken bones and organ health
add: Organ Smartfridges! Store your organs safely. Upgrading the matter bin will give it the ability to heal organs held within- though this does not repair organs that are currently failing, however.
add: Lobectomy! Not lobotomy, but cuts out a damaged lung lobe to make its performance passable. Restores a lung to working order if it was too damaged to work before, but can only be done once. The lobectomy step has a 95% chance of success, so use some form of sterilizer to ensure you don't accidentally get blood inside your patient's lungs!
add: Coronary Bypass! Restores a heart to working order if it was too damaged to work before, but can only be done once. Both incising the heart and grafting the bypass have a 90% chance to succeed, and failing them will cause heart damage and massive amounts of bleeding- so be sure to work with ideal conditions, proper tools, and sterilizer!
add: Organs now decay when not in a living host
add: Severe heart damage increases breathing rate, and causes a heart attack on failure
add: Severe lung damage increases breathing rate, and causes suffocation on failure
add: Stomach damage will likely cause the owner to vomit depending on how injured the stomach is, and how much they've eaten
tweak: Brain damage effects now works similarly to the rest of the organ damage
tweak: Appendicitis will do appendix damage when in stages 3 and 4, which can cause it to burst if left
untreated
tweak: Organs heal faster the healthier the host is
balance: Defib timer has been extended to 15 minutes
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
```
proc name: on life (/obj/item/organ/proc/on_life)
src: the plasma bone \"tongue\" (/obj/item/organ/tongue/bone/plasmaman)
src.loc: null
call stack:
the plasma bone \"tongue\" (/obj/item/organ/tongue/bone/plasmaman): on life()
Redacted (/mob/living/carbon/human): handle organs()
Redacted (/mob/living/carbon/human): Life(2, 3)
Redacted (/mob/living/carbon/human): Life(2, 3)
Redacted (/mob/living/carbon/human): Life(2, 3)
Mobs (/datum/controller/subsystem/mobs): fire(0)
Mobs (/datum/controller/subsystem/mobs): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop()
Master (/datum/controller/master): StartProcessing(0)
```

**Comment by AffectedArc07** [comment]
Will probably close in favour of https://github.com/KeplerStation/KeplerStation/pull/703

Its been better tested upstream and probably has addins for citadel changes

**Comment by AffectedArc07** [comment]
Closing from remake
